internal/openfga/user.go

```diff
@@ -280,6 +282,46 @@ func (u *User) ListApplicationOffers(ctx context.Context, relation ofga.Relation
 	return appOfferUUIDs, err
 }
 
+// CheckPermission returns an Unauthorized error if user is not allowed specified permission to the resource.
+func (u *User) CheckPermission(ctx context.Context, resourceTag string, permission string) error {
+	resource, err := names.ParseTag(resourceTag)
+	if err != nil {
+		return err
+	}
+
+	var relation Relation
+	switch permission {
+	case "superuser", string(params.ModelAdminAccess):
+		relation = ofganames.AdministratorRelation
+	case string(params.ModelWriteAccess):
+		relation = ofganames.WriterRelation
+	case string(params.ModelReadAccess):
+		relation = ofganames.ReaderRelation
+	case string(params.OfferConsumeAccess):
+		relation = ofganames.ConsumerRelation
+	case string("add-cloud"):
+		relation = ofganames.CanAddModelRelation
```

Comment:
I think we already have a function that converts Juju permissions to OpenFGA relations

Reply:
in openfga/names we have the inverse function, but not this

Reply:
Had to look a bit but I think there is logic like this in `internal/jimm/permissions/access.go`, see `ToCloudRelation`, `ToModelRelation`, etc. Possibly those function should be here or something like that.

Reply:
we'll have an import cycle if i use methods from internal/jimm/permissions

Reply:
That's fine, can you make a follow-up to put this logic in a separate function and use it from `internal/jimm/permissions`? Or just look at whether we should at the least.